### PR TITLE
Add logs support on Kafka receiver

### PR DESCRIPTION
**Pull request by sincejune**
**Description:**
Add logs support on Kafka exporter

**Link to tracking Issue:**
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/1331
Related to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/268
Related to https://github.com/open-telemetry/opentelemetry-collector/pull/2943

**Testing:**
- Added corresponding unit tests according to traces and metrics.
- Tested locally with Kafka 2.11-2.0.0

**Documentation:**
Update Kafka receiver README and Receiver README.

### CODE REVIEW

File: receiver/kafkareceiver/unmarshaller.go [hunk 1]
```diff
@@ -27,6 +27,15 @@ type Unmarshaller interface {
 	Encoding() string
 }
 
+// logsUnmarshaller deserializes the message body.
+type logsUnmarshaller interface {
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/unmarshaller.go:31]: If this is private means you don't expect users to implement? Also probably we should rename the previous one to TracesUnmarshaller.

File: receiver/kafkareceiver/factory.go [hunk 2]
```diff
@@ -53,18 +53,30 @@ func WithAddUnmarshallers(encodingMarshaller map[string]Unmarshaller) FactoryOpt
 	}
 }
 
+// WithAddLogsUnmarshallers adds logsUnmarshallers.
+func WithAddLogsUnmarshallers(encodingMarshaller map[string]logsUnmarshaller) FactoryOption {
```
Comment (bogdandrutu) [inline receiver/kafkareceiver/factory.go:57]: Names are inconsistent, probably worth updating the previous one to include traces.

Also I would change the API to:

```golang
func WithLogsUnmarshallers(encoding string, unmarshaller logsUnmarshaller) FactoryOption {}
```

Or even consider to add a `name` func on the interface to simplify usage.

Comment (bogdandrutu) [reply receiver/kafkareceiver/factory.go:57]: Actually there is an `Encoding` so we can do:

```golang
func WithLogsUnmarshallers(unmarshallers logsUnmarshaller...) FactoryOption {}
```

File: receiver/kafkareceiver/unmarshaller.go (hunk 1 above)
Comment (sincejune) [reply receiver/kafkareceiver/unmarshaller.go]: Thanks, @bogdandrutu. Fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/2944/commits/0cb875fe14ee9e4b8def323d97a84c08d1ffec5a

And actually, there're many methods to be renamed (adding `Traces`). I would like to do this in another PR cuz it's already not a small one.

File: receiver/kafkareceiver/factory.go (hunk 2 above)
Comment (sincejune) [reply receiver/kafkareceiver/factory.go]: The method is actually consistent with the traces' method here: https://github.com/open-telemetry/opentelemetry-collector/blob/6458914bbd0d06bc3fa6304197fe736d47b0f694/receiver/kafkareceiver/factory.go#L47-L54

I'd like to keep the parameters consistent with trace's method and there are multiple encodings and return one factory in traces' method. But I'd like to rename the method name to be consistent in a subsequent PR. Thoughts?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**